cmd/vpcgen/main.go

```diff
@@ -9,229 +9,18 @@ VpcGen reads specification files adhering to spec_schema.json and generates netw
-	}
-	return nil
-}
-
-func writeOutput(collection ir.Collection, defs *ir.ConfigDefs, outputDirectory, outputFormat, outputFile,
-	prefixOfFileNames, configFilename string) {
-	if outputDirectory == "" {
-		writeToFile(collection, "", outputFormat, outputFile, configFilename)
-	} else {
-		// create the directory if needed
-		err := os.MkdirAll(outputDirectory, defaultDirectoryPermission)
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		// write each file
-		for vpc := range defs.VPCs {
-			suffix := vpc + "." + outputFormat
-			outputPath := outputDirectory + "/" + suffix
-			if prefixOfFileNames != "" {
-				outputPath = outputDirectory + "/" + prefixOfFileNames + "_" + suffix
-			}
-			writeToFile(collection, vpc, outputFormat, outputPath, configFilename)
-		}
-	}
-}
-
-func writeToFile(collection ir.Collection, vpc, outputFormat, outputFile, configFilename string) {
-	var data bytes.Buffer
-	writer, err := pickWriter(outputFormat, &data, configFilename)
-	if err != nil {
-		log.Fatal(err)
-	}
-	if err = collection.Write(writer, vpc); err != nil {
-		log.Fatal(err)
-	}
-
-	if outputFile == "" {
-		fmt.Print(data.String())
-	} else {
-		err = os.WriteFile(outputFile, data.Bytes(), defaultFilePermission)
-		if err != nil {
-			log.Fatal(err)
-		}
-	}
-}
-
 func main() {
-	configFilename := flag.String("config", "",
-		"JSON file containing config spec")
-	target := flag.String("target", defaultTarget,
-		fmt.Sprintf("Target resource to generate. One of %q, %q, %q.", aclTarget, sgTarget, singleaclTarget))
-	outputFormat := flag.String("fmt", "",
-		fmt.Sprintf("Output format. One of %q, %q, %q; must not contradict output file suffix. (default %q)",
-			tfOutputFormat, csvOutputFormat, mdOutputFormat, defaultOutputFormat))
-	outputFile := flag.String("output-file", "",
-		"Write all generated resources to the specified file.")
-	outputDirectory := flag.String("output-dir", "",
-		"Write generated resources to files in the specified directory, one file per VPC.")
-	prefixOfFileNames := flag.String("prefix", "", "The prefix of the files that will be created.")
-	flag.Usage = func() {
-		_, _ = fmt.Fprintf(os.Stderr, `VpcGen translates connectivity spec to network ACLs or Security Groups.
-Usage:
-	%s [flags] SPEC_FILE
-
-SPEC_FILE: JSON file containing connectivity spec, and segments.
-
-Flags:
-`, "vpcgen")
-		flag.PrintDefaults()
-	}
-	flag.Parse()
-
-	connectivityFilename := flag.Arg(0)
-	if connectivityFilename == "" || flag.NArg() != 1 {
-		flag.Usage()
-		os.Exit(0)
-	}
-
-	var err error
-
-	if *configFilename == "" {
-		log.Fatal("-config parameter must be supplied")
-	}
+	rootCmd := subcmds.NewRootCommand()
+	rootCmd.SetArgs(os.Args[1:])
 
-	if *outputDirectory != "" && *outputFile != "" {
-		log.Fatal(fmt.Errorf("output-file and output-dir cannot be specified together"))
-	}
-
-	*outputFormat, err = pickOutputFormat(*outputFormat, *outputFile, *outputDirectory)
+	err := rootCmd.Execute()
 	if err != nil {
 		log.Fatal(err)
 	}
```

Comment:
Errors are currently printed twice - by Cobra and here. Either remove the printing to log here, or suppress errors in Cobra.